backend/src/controllers/reports/helpers/index.js

```diff
@@ -57,36 +71,59 @@ const getDocumentApiBody = async (
   };
 };
 
+// helper functions for getDocumentApi()
+const isValidInput = ({ templateType, templateFileName }) =>
+  templateFiletypes.includes(templateType) &&
+  templateFiletypes.includes(templateFileName.slice(-4));
+
+const getTemplatePath = ({ templateType, templateFileName }) =>
+  `../../../../reports/${templateType}/${templateFileName}`;
+
+// decide whether to export an XLSX or PDF
+const getOutputFormat = (templateType) =>
+  ({
+    docx: "pdf",
+    xlsx: "xlsx",
+  }[templateType]);
+
+// apply headers to the request, then return the data
 const getReport = async (request, reply) => {
+  applyRequestHeaders(reply);
+  return request.data;
+};
+
+const applyRequestHeaders = (reply) =>
   reply.type("application/pdf").headers({
     "Content-Disposition": 'attachment;filename="test.pdf"',
   });
 
-  return request.data;
-};
-
 /**
- * Separates an array of projects into groups by property.
+ * Groups an array of objects by a specified property.
  *
- * this moves each array of projects onto "projects" key, and the property onto a {property} key
- * the Carbone.io engine needs to see objects nested in arrays, or arrays nested in objects, or else it can't iterate through.
- * arrays inside of arrays, or objects inside of objects can't be iterated properly (at depth n+1)
- *
- * @param   {any[]}   rows Array of projects ordered by some property.
- * @param   {string}  prop string matches property name we want to group the rows by
- * @returns {any[][]}
+ * @param   {object[]} rows - The array of objects to group.
+ * @param   {string}   prop - The property to group by.
+ * @returns {object[]}      Returns an array of objects grouped by the specified property.
  */
 const groupByProperty = (rows, prop) => {
```

Comment:
is the only difference here, is that array of objects, or have you changed something, as this function is used in other locations. again this would be a perfect function for unit tests.